balancer/rls/balancer_test.go

```diff
@@ -652,6 +652,192 @@ func (s) TestConfigUpdate_DataCacheSizeDecrease(t *testing.T) {
 	verifyRLSRequest(t, rlsReqCh, true)
 }
 
+// Test that when a data cache entry is evicted due to config change
+// in cache size, the picker is updated accordingly.
+func (s) TestPickerUpdateOnDataCacheSizeDecrease(t *testing.T) {
+	// Create a restartable listener which can close existing connections.
+	l, err := testutils.LocalTCPListener()
+	if err != nil {
+		t.Fatalf("net.Listen() failed: %v", err)
+	}
+	lis := testutils.NewRestartableListener(l)
```

Comment:
We don't need a restartable listener for this test. Is there a reason to use this? Instead, you could simply pass `nil` to: `	rlsServer, rlsReqCh := rlstest.SetupFakeRLSServer(t, nil)`

Reply:
Yeah my bad it was removed in earlier commits, this got back while resolving merge conflicts. Updated it now.